go-controller/pkg/util/pod_annotation.go

```diff
@@ -597,3 +608,35 @@ func AddRoutesGatewayIP(
 
 	return nil
 }
+
+// UnmarshalUDNOpenPortsAnnotation returns the OpenPorts from the pod annotation. If annotation is not present,
+// empty list with no error is returned.
+func UnmarshalUDNOpenPortsAnnotation(annotations map[string]string) ([]*OpenPort, error) {
+	result := []*OpenPort{}
+	ports, ok := annotations[UDNOpenPortsAnnotationName]
+	if !ok {
+		return result, nil
+	}
+	for _, portDef := range strings.Split(ports, ",") {
+		parsedPort := strings.Split(portDef, ":")
+		if len(parsedPort) != 2 {
+			return nil, fmt.Errorf("failed to parse port definition %q", portDef)
+		}
+		protocol := strings.ToLower(parsedPort[0])
+		if protocol != "tcp" && protocol != "udp" && protocol != "sctp" {
```

Comment:
what if a user wants to specify a protocol other than these 3? Let's say a user just wants to be able to ping a pod with ICMP. Should we allow a user to just match on port if protocol is empty? or does ovn support matching on ICMP as well?

Reply:
We can add icmp match, but I am not sure why anyone may need to just ping a pod, even for some liveness check I would expect something more than just a ping.
I didn't get the point about matching on port without protocol - how is that expected to work?

Reply:
we have had some users in the past that just want to use icmp to make sure their pod is alive. I guess we cannot match on just port without specifying protocol. Let's see if we can include ICMP and if there are any other protocols missing that OVN supports matching on.